File: sandbox-runner/patterns/cancellation-propagation/java/src/main/java/Starter.java
import io.temporal.client.WorkflowClient;
import io.temporal.client.WorkflowOptions;
import io.temporal.client.WorkflowStub;
import io.temporal.serviceclient.WorkflowServiceStubs;

public class Starter {
    public static void main(String[] args) throws InterruptedException {
        WorkflowServiceStubs service = WorkflowServiceStubs.newLocalServiceStubs();
        WorkflowClient client = WorkflowClient.newInstance(service);

        String workflowId = Shared.WORKFLOW_ID_PREFIX + "-" + System.currentTimeMillis();
        String orderId = "order-42";

        CancellationPropagationWorkflow.Parent workflow =
                client.newWorkflowStub(
                        CancellationPropagationWorkflow.Parent.class,
                        WorkflowOptions.newBuilder()
                                .setTaskQueue(Shared.TASK_QUEUE)
                                .setWorkflowId(workflowId)
                                .build());

        // Start asynchronously so the running workflow can be signalled.
        WorkflowClient.start(workflow::run, orderId);
        System.out.println("Started workflow: " + workflowId);
        System.out.println(
                "Fulfilling " + orderId + "; children are reserving resources concurrently…");

        // Let the children apply their steps, then request a stop.
        Thread.sleep(2000);
        System.out.println("Requesting stop; cancellation will propagate to every child…");
        workflow.stop();

        String result = WorkflowStub.fromTyped(workflow).getResult(String.class);
        System.out.println(result);
        System.out.println(
                "Open the Temporal UI and search for '" + workflowId
                        + "' to see each child transition to Canceled after compensating.");
    }
}

File: sandbox-runner/patterns/cancellation-propagation/go/starter.go
package main

import (
	"context"
	"fmt"
	"log"
	"time"

	"go.temporal.io/sdk/client"
)

func main() {
	c, err := client.Dial(client.Options{HostPort: "localhost:7233"})
	if err != nil {
		log.Fatalln("Unable to create client:", err)
	}
	defer c.Close()

	ctx := context.Background()
	workflowID := fmt.Sprintf("%s-%d", WorkflowIDPrefix, time.Now().UnixMilli())
	orderID := "order-42"

	we, err := c.ExecuteWorkflow(ctx, client.StartWorkflowOptions{
		ID:        workflowID,
		TaskQueue: TaskQueue,
	}, FulfillOrderWorkflow, orderID)
	if err != nil {
		log.Fatalln("Unable to execute workflow:", err)
	}
	fmt.Printf("Started workflow: %s\n", we.GetID())
	fmt.Printf("Fulfilling %s; children are reserving resources concurrently…\n", orderID)

	// Let the children apply their steps, then request a stop.
	time.Sleep(2 * time.Second)
	fmt.Println("Requesting stop; cancellation will propagate to every child…")
	if err := c.SignalWorkflow(ctx, workflowID, we.GetRunID(), "stop", nil); err != nil {
		log.Fatalln("Unable to signal workflow:", err)
	}

	var result string
	if err := we.Get(ctx, &result); err != nil {
		log.Fatalln("Workflow failed:", err)
	}
	fmt.Println(result)
	fmt.Printf(
		"Open the Temporal UI and search for '%s' to see each child transition to Canceled after compensating.\n",
		workflowID,
	)
}
